pages/_app.tsx

```diff
@@ -1,9 +1,46 @@
-// import '../styles/globals.css'
-import 'tailwindcss/tailwind.css'
+import '../styles/globals.css'
+// import 'tailwindcss/tailwind.css'
 import type { AppProps } from 'next/app'
+import { ConnectionProvider, WalletProvider } from '@solana/wallet-adapter-react'
+import { WalletModalProvider } from '@solana/wallet-adapter-react-ui'
+import { WalletAdapterNetwork } from '@solana/wallet-adapter-base'
+import { clusterApiUrl } from '@solana/web3.js'
+import {
+  GlowWalletAdapter,
+  PhantomWalletAdapter,
+  SlopeWalletAdapter,
+  SolflareWalletAdapter,
+  TorusWalletAdapter,
+} from "@solana/wallet-adapter-wallets";
+
+// Default styles for wallet adapter
+require('@solana/wallet-adapter-react-ui/styles.css');
 
 function MyApp({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+
+  const network = WalletAdapterNetwork.Devnet;
```

Comment:
This has to come from the ENV, so it can be updated without touching the code. 

WalletAdapterNetwork is just a Constance, so you can have  `CLUSTER_ENDPOINT=https://api.devnet.solana.com` as env